src/gluonts/model/wavenet/_estimator.py

```diff
@@ -269,9 +271,11 @@ def train(self, training_data: Dataset) -> Predictor:
         # Here we override the prediction length for training.
         # This computes the loss over longer windows and makes the convolutions more
         # efficient, since calculations are reused.
-        pred_length = min(mean_length, self.train_window_length)
+        pred_length = min(
+            median_length - self.context_length, self.train_window_length
```

Comment:
This can cause issues when `context_length` is larger than `median_length`.

Reply:
Added a guard to prevent negative prediction length.